exporter/opentelemetry-exporter-otlp-proto-common/src/opentelemetry/exporter/otlp/proto/common/_internal/__init__.py

```diff
@@ -151,6 +152,27 @@ def _encode_attributes(
     return pb2_attributes
 
 
+class ThreadWithReturnValue(threading.Thread):
+    def __init__(
+        self,
+        group=None,
+        target=None,
+        name=None,
+        args=(),
+        kwargs=None,
+    ):
+        threading.Thread.__init__(self, group, target, name, args, kwargs)
+        self._return = None
+
+    def run(self):
+        if self._target is not None:  # type: ignore
+            self._return = self._target(*self._args, **self._kwargs)  # type: ignore
+
+    def join(self, *args):  # type: ignore
+        threading.Thread.join(self, *args)
```

Comment:
nit: Could we avoid type ignore by explicitly passing the expected type?
```suggestion
    def join(self, timeout: float | None = None) -> Any:
        threading.Thread.join(self, timeout=timeout)
```